crates/apollo-router-core/src/query_planner/model.rs

```diff
@@ -196,11 +196,7 @@ impl PlanNode {
                     .instrument(tracing::info_span!("fetch"))
                     .await
                     {
-                        Ok(()) => {
-                            let received =
-                                serde_json::to_string_pretty(&response.lock().await.data).unwrap();
-                            tracing::trace!("New data:\n{}", received,);
-                        }
```

Comment:
This was actually useful for me for debugging the integration tests. Would it be possible to have it in debug build? (ofc without the string_pretty XD `{:#?}` is enough)

Reply:
Even with {:?} it's not the right place to put it, because of the lock and await that interact badly with tracing. It might be better to log in the subgraph fetch code, but I suspect you could do it with a log filter in cli, since reqwest is probably logging something

Reply:
Logging the output of the subgraph is not good enough I think. The idea was to see the "progression" of the `data` getting filled. Maybe I don't need this anymore anyway.

Reply:
I see, that's definitely useful. In `insert_data` then? Or maybe we can look for yet another way to remove the mutex 😁